### ISSUE TITLE
Compile to WASM with p_threads

### ISSUE THREAD

**Issue by Myrindd** [jrouwe/JoltPhysics.js#10]
I am trying to compile to WASM with p_threads in order to use as a library inside a larger WASM project that is multithreaded.
In the cmakelists, I added "-s USE_PTHREADS" in order to make the compilation work for the larger project. Afterward, I simply replicated the helloworld example, but when creating a shape with the body interface, the application crashes with the traditional "memory access out of bound" error happening during atomic_fetch_add. Is it possible to compile with p_thread to WASM? If so, how should the cmakelist be modified?
As a side node, the compiling of jolt seems to fail in "release" mode.

**Comment by Myrindd** [comment]
Seems like it might not be related to p_threads... could you please share the cmakelist to compile to WASM?

**Comment by jrouwe** [comment]
Do you have a full callstack for this? And are there any interesting things in the console output (e.g. it's running out of memory?)

**Comment by Myrindd** [comment]
Here is the result in the console:
<img width="494" alt="image" src="https://github.com/jrouwe/JoltPhysics.js/assets/67431109/88a799b2-0439-4c4a-b417-129af99d0cca">

**Comment by Myrindd** [comment]
Here is the result of -fsanitize=address
<img width="796" alt="image" src="https://github.com/jrouwe/JoltPhysics.js/assets/67431109/bdd5bef8-eec5-4cd5-ab69-62948cd665b6">

**Comment by jrouwe** [comment]
According to ASAN your stack is not big enough. I have ```-s TOTAL_STACK=1048576``` on my emcc commandline to enlarge the stack size.

PhysicsSystem (1.3K) and JobSystemThreadPool (8.6K) are large classes. If you don't want to allocate more stack, you can also new these so they end up on the heap instead of on the stack.

**Comment by jrouwe** [comment]
The floor_settings object is 240 bytes in single precision mode and 256 bytes in double precision mode. The only difference is in the mPosition member (which is at the beginning of the class). This does not seem to match up with the 'offset 500' that ASAN is complaining about.

Could it be that you compile the Jolt source code with a different set of defines than the rest of your code? E.g. if you compile Jolt with JPH_DOUBLE_PRECISION then Jolt would treat the struct as 256 bytes long and if you compile your own code without JPH_DOUBLE_PRECISION then it would treat it as a 240 bytes object so it would contain garbage for Jolt.

**Comment by Myrindd** [comment]
I am compiling Jolt as a .a shared library that I incorporate in the cmakelist for my own system; but I also refer to the actual source code, so it could be indeed. I will have a look.

**Comment by Myrindd** [comment]
You are absolutely right; it appears that was the issue. Sorry for wasting your time! Amazing job by the way; as someone who has developed a functional 3D physics engine before, I understand how painstakingly difficult it is (which is why I am considering using yours!).
